contrib/net/http/trace.go

```diff
@@ -41,20 +42,33 @@ type ServeConfig struct {
 // TraceAndServe serves the handler h using the given ResponseWriter and Request, applying tracing
 // according to the specified config.
 func TraceAndServe(h http.Handler, w http.ResponseWriter, r *http.Request, cfg *ServeConfig) {
-	if cfg == nil {
-		cfg = new(ServeConfig)
+	span, ctx := StartRequestSpan(r, cfg.Service, cfg.Resource, cfg.QueryParams, cfg.SpanOpts...)
+	rw, ddrw := wrapResponseWriter(w)
+	defer func() {
+		FinishRequestSpan(span, ddrw.status, cfg.FinishOpts...)
+	}()
+
+	if appsec.Enabled() {
+		h = httpsec.WrapHandler(h, span, cfg.RouteParams)
 	}
+	h.ServeHTTP(rw, r.WithContext(ctx))
+}
+
+// StartRequestSpan starts an HTTP request span with the standard list of HTTP request span tags. URL query parameters
```

Comment:
What is the "standard list of HTTP request span tags". Is that documented somewhere? If so, link to that here. If not, I suggest writing them out explicitly or providing more explanation in the godoc of this function.

Reply:
The document is internal in our confluence: https://datadoghq.atlassian.net/wiki/spaces/APM/pages/2357395856/Span+attributes

What I personally think would be cleaner and better would be to move all the HTTP tag names currently defined in `ddtrace/ext/tags.go` into a const block of the new internal httptrace package so that users of this package can clearly see their list and documentation:

```go
// List of standard HTTP request span tags
const (
	// Method of the HTTP request.
	Method = "http.method"
	// StatusCode is the client the response status code the HTTP client received or the HTTP server sent.
	StatusCode = "http.status_code"
	// ... etc ...
)
```

Reply:
SGTM 👍

Reply:
Done in https://github.com/DataDog/dd-trace-go/pull/1286/commits/e989bc6a2b14146ed4a2186b599b1aed413c7699
But note that I couldn't move the existing public definitions of the `ext` package to avoid breaking changes and so I redefinted them in httptrace for the sake of creating this documentation group.